moveit_core/collision_detection_fcl/src/collision_common.cpp

```diff
@@ -404,29 +405,33 @@ bool distanceCallback(fcl::CollisionObject* o1, fcl::CollisionObject* o2, void*
             cd2->ptr.link :
             (cd2->type == BodyTypes::ROBOT_ATTACHED ? cd2->ptr.ab->getAttachedLink() : nullptr);
 
+    if (!l1 || cdata->req->active_components_only->find(l1) == cdata->req->active_components_only->end())
+      active1 = false;
+
+    if (!l2 || cdata->req->active_components_only->find(l2) == cdata->req->active_components_only->end())
+      active2 = false;
+
     // If neither of the involved components is active
-    if ((!l1 || cdata->active_components_only_->find(l1) == cdata->active_components_only_->end()) &&
-        (!l2 || cdata->active_components_only_->find(l2) == cdata->active_components_only_->end()))
+    if (active1 == false && active2 == false)
     {
-      min_dist = cdata->res_->distance;
-      return cdata->done_;
+      return false;
```

Comment:
Why did you dropped setting `min_dist`? Probably, that's fine. I just don't understand the relevance of the previous code. I'm not even sure that the right-hand value had any valid initialization before.

Reply:
The min_dist does the same thing as distance_threshold in the distance request when searching for global minimum. Now that we enable the ability to get not only the global minimum but the ability to get information about every link, the information is now stored on a link by link bases in the distance request.